routes/score.router.js

```diff
@@ -0,0 +1,10 @@
+const express = require("express");
+const router = express.Router();
+const { addScore, getUserDashboard } = require("../controllers/score.controller");
+const { checkAuth } = require("../middlewares/checkAuth");
+
+router.route("/:userId/get-user-dashboard").all(checkAuth).get(getUserDashboard);
```

Comment:
What does this `.all()` function do? Does it behave like middleware?

Reply:
Yep, It is kind of a middleware which checks some condition on all the methods which are passing through this route.
For. eg., Here only the get method is present for this route if I add a post method here then that method also will be checked by this ```.all()``` middleware.